### ISSUE TITLE
Update references to Twitter/X everywhere

### ISSUE THREAD

**Issue by felipevelzani** [WordPress/gutenberg#55142]
## What problem does this address?

We need to update the references (name and logo) from Twitter to X everywhere to reflect their new branding.

Note: Honestly, I don't know if they even decided on a final logo, but I feel like we should start addressing it from our end.

This was originally reported on WP.org forums: https://wordpress.org/support/topic/new-twitter-x-logo-2/?view=all

## What is your proposed solution?

Basically, add their new logo/name everywhere. Some examples are the Social Icons block and the embed feature.

**Comment by sabernhardt** [comment]
#54092 updated the Social block, but [Core Trac 61050](https://core.trac.wordpress.org/ticket/61050) calls attention to updating the Embed block.

**Comment by WraithKenny** [comment]
It's cool how if this block is used on a client site, and they ask for the X logo, we have to delete the block, and construct the whole thing from scratch manually.
